Releases/liquid.cache.sqlserver v2.0.0 preview

lucianareginalino opened the pull request:
This implementation just extends  Microsoft.Extensions.Caching.SqlServer DI by including Liquid.Cache decorator registry.
Resolves https://github.com/Avanade/Liquid-Application-Framework/issues/144

guilhermeluizsp commented on src/Liquid.Base/WorkbenchServiceHttp.cs:
```diff
@@ -0,0 +1,56 @@
+        POST,
+
+        /// <summary>
+        /// Represents an HTTP PUT protocol method that is used to post a new entity as an addition to a URI.
```
I believe the docs for this and the other items below are incorrect. Not all values are used to post a new entity, right?

bruno-brant replied:
Yeah, actually, the text is very bad. I'll improve it.

guilhermeluizsp commented on src/Liquid.Base/WorkbenchServiceType.cs:
```diff
@@ -0,0 +1,87 @@
+        Repository,
+
+        /// <summary>
+        /// Service used to record events and general data in telemetry
```
It's a good practice to end documentation phrases with a period (.).

bruno-brant replied:
Ooops

guilhermeluizsp commented on src/Liquid.Base/WorkbenchServiceType.cs:
```diff
@@ -0,0 +1,87 @@
+        Repository = WorkbenchServiceType.Repository,
+
+        /// <summary>
+        /// Service used to record events and general data in telemetry
```
Same as above. It should end with a period (.).

guilhermeluizsp commented on test/Liquid.Base.Tests/WorkbenchTests.cs:
```diff
@@ -0,0 +1,425 @@
+        }
+
+        [Theory, AutoSubstituteData]
+        public void GetRegisteredServiceReturnsWhatWasAddedByAddToCache(WorkbenchServiceType type, IWorkBenchService service)
```
Is there any specific reason for this test to use the misspelled `IWorkBenchService`?

bruno-brant replied:
yes: a mistake 😁

Actually, originally, I was going to leave the tests on the old version in order to prove that old functionality still held. But I ended up moving the tests but for some reason, it seems, I stopped in the middle of it.

guilhermeluizsp commented on test/Liquid.Base.Tests/WorkbenchTests.cs:
```diff
@@ -0,0 +1,425 @@
+        }
+
+        [Theory, AutoSubstituteData]
+        public void AddToCacheSameServiceTypeTwiceThrows(WorkbenchServiceType type, IWorkBenchService service1, IWorkBenchService service2)
```
Same thing as above. It should work fine with the new correctly spelled interface.

guilhermeluizsp commented on test/Liquid.Base.Tests/WorkbenchTests.cs:
```diff
@@ -0,0 +1,425 @@
+        }
+
+        [Theory, AutoSubstituteData]
+        public void GetRegisteredServiceCallsInitializeOnService(WorkbenchServiceType type, IWorkBenchService service)
```
Again. We should move on to using the new interface to accelerate the obsolescence of the old one.

guilhermeluizsp commented:
There are some breaking changes since you've renamed some interfaces (`ILightLogger` inherits from `IWorkbenchHealthCheck` now, instead of `IWorkBenchHealthCheck`, for example), but I believe they are necessary, considering the goal of this pull request.

Everything looks great, but both `ILightPolly` and `ILightWorker` still inherits from the old `IWorkBenchService` interface (the misspelled one). They don't show up in the file changes because you haven't touched them, but I think they should be using the new one.

https://github.com/Avanade/Liquid-Application-Framework/blob/939759831f202b748565a454a2828477df4d0c45/src/Liquid.Base/Interfaces/Polly/ILightPolly.cs#L10

https://github.com/Avanade/Liquid-Application-Framework/blob/939759831f202b748565a454a2828477df4d0c45/src/Liquid.Base/Interfaces/MessageBus/ILightWorker.cs#L8

guilhermeluizsp commented:
@bruno-brant I'm waiting for you to merge with the master branch since #141 is approved already.

bruno-brant commented:
> There are some breaking changes since you've renamed some interfaces (`ILightLogger` inherits from `IWorkbenchHealthCheck` now, instead of `IWorkBenchHealthCheck`, for example), but I believe they are necessary, considering the goal of this pull request.
> 
> Everything looks great, but both `ILightPolly` and `ILightWorker` still inherits from the old `IWorkBenchService` interface (the misspelled one). They don't show up in the file changes because you haven't touched them, but I think they should be using the new one.
> 
> https://github.com/Avanade/Liquid-Application-Framework/blob/939759831f202b748565a454a2828477df4d0c45/src/Liquid.Base/Interfaces/Polly/ILightPolly.cs#L10
> 
> https://github.com/Avanade/Liquid-Application-Framework/blob/939759831f202b748565a454a2828477df4d0c45/src/Liquid.Base/Interfaces/MessageBus/ILightWorker.cs#L8

Yeah, I realized that the `IWorkbenchHealthCheck` change is breaking, but I also guessed that it wouldn't be a big problem for users since, well, they shouldn't be accessing a `ILightLogger` cast to a `IWorkbenchHealthCheck`. I know that it's a possibility, but it's one I'm okay with.

Now, both polly and worker are just things that I ended up forgetting. I'll make the necessary fixes.

bruno-brant commented:
Also, both Polly and Worker will be made obsolete very soon. :)

bruno-brant commented:
Done, @guilhermeluizsp. The branch is rebased.

lucianareginalino committed:
feat(Liquid.Cache.SqlServer): Build Liquid.Cache.SqlServer package.
This implementation just extends  Microsoft.Extensions.Caching.SqlServer DI by including Liquid.Cache decorator registry.
Resolves #144
Signed-off-by: Luciana Regina Lino <luciana.regina.lino@avanade.com>

lucianareginalino committed:
feat(liquid-ci-cd-cache-sqlserver.yml): Creates Liquid.Cache.Memory CI/CD Action.
Signed-off-by: Luciana Regina Lino <luciana.regina.lino@avanade.com>
